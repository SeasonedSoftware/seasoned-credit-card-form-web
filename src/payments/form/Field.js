import React, { Fragment } from 'react'
import { Field } from 'formik'
import InputMask from 'react-input-mask'
import Media from 'react-media'

import {
  FormControl,
  InputLabel,
  Input,
  FormHelperText,
} from '@material-ui/core'

export default props => {
  const { errors, touched, name, mask, maskChar, placeholder } = props
  const error = touched && touched[name] && errors && errors[name]

  return (
    <FormControl
      component="fieldset"
      id={name + '-fieldset'}
      disabled={props.disabled}
      error={!!error}
      fullWidth={props.fullWidth}
    >
      <Media query="(max-width: 425px)">
        <Fragment>
          <Field
            type={props.type}
            name={name}
            onChange={props.handleChange}
            onBlur={props.handleBlur}
            onFocus={props.onFocus}
            disabled={props.disabled}
            id={name}
            value={props.value}
            component={Input}
            inputComponent={mask ? InputMask : undefined}
            inputProps={mask ? { mask, maskChar } : {}}
            placeholder={
              props.placeholder
                ? `${props.label} ${props.placeholder}`
                : props.label
            }
            error={!!error}
            autoFocus={props.autoFocus}
          />
          {error && <FormHelperText>{error || placeholder}</FormHelperText>}
        </Fragment>
      </Media>
      <Media query="(min-width: 426px)">
        <Fragment>
          <InputLabel htmlFor={name}>{props.label}</InputLabel>
          <Field
            type={props.type}
            name={name}
            onChange={props.handleChange}
            onBlur={props.handleBlur}
            onFocus={props.onFocus}
            id={name}
            value={props.value}
            component={Input}
            inputComponent={mask ? InputMask : undefined}
            inputProps={mask ? { mask, maskChar } : {}}
            placeholder={props.placeholder}
            error={!!error}
            autoFocus={props.autoFocus}
          />
          <FormHelperText>{error || placeholder}</FormHelperText>
        </Fragment>
      </Media>
    </FormControl>
  )
}
