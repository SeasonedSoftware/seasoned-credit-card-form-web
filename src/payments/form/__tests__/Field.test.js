import React from 'react'
import renderer from 'react-test-renderer'
import Field from '../Field'

jest.mock('react-media', () => props => <div>{props.children}</div>)
jest.mock('formik', () => ({
  Field: props => <div {...props}>Field</div>,
}))

it('renders correctly without errors', () => {
  const tree = renderer.create(<Field />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with errors', () => {
  const props = {
    name: 'abc',
    touched: {
      abc: 'X',
    },
    errors: {
      abc: 'Y',
    },
  }

  const tree = renderer.create(<Field {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
