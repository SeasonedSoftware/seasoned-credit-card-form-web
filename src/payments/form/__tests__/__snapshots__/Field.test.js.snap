// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with errors 1`] = `
<fieldset
  className="MuiFormControl-root-1"
  id="abc-fieldset"
>
  <div>
    <div
      component={[Function]}
      error={true}
      id="abc"
      inputProps={Object {}}
      name="abc"
    >
      Field
    </div>
    <p
      className="MuiFormHelperText-root-23 MuiFormHelperText-error-24"
    >
      Y
    </p>
  </div>
  <div>
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-error-19 MuiInputLabel-error-8 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="abc"
    />
    <div
      component={[Function]}
      error={true}
      id="abc"
      inputProps={Object {}}
      name="abc"
    >
      Field
    </div>
    <p
      className="MuiFormHelperText-root-23 MuiFormHelperText-error-24"
    >
      Y
    </p>
  </div>
</fieldset>
`;

exports[`renders correctly without errors 1`] = `
<fieldset
  className="MuiFormControl-root-1"
  id="undefined-fieldset"
>
  <div>
    <div
      component={[Function]}
      error={false}
      inputProps={Object {}}
    >
      Field
    </div>
  </div>
  <div>
    <label
      className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
    />
    <div
      component={[Function]}
      error={false}
      inputProps={Object {}}
    >
      Field
    </div>
    <p
      className="MuiFormHelperText-root-23"
    />
  </div>
</fieldset>
`;
